diff --git a/dist/lib/fcm.js b/dist/lib/fcm.js
index 2909be61006e96a822145e7ff52315b6a6ad1de5..d6e22872e4627f162bc8d3976dced96cc64aab73 100644
--- a/dist/lib/fcm.js
+++ b/dist/lib/fcm.js
@@ -268,6 +268,9 @@ class FCM {
                     await new Promise((resolve) => setTimeout(resolve, 1000 * (tries + 1)));
                     return this.sendRequest(device, message, projectId, accessToken, tries + 1);
                 }
+                else if (response.status === 404) {
+                    throw new Error("UNREGISTERED");
+                }
                 else if (response.status === 400 &&
                     data.error &&
                     data.error.message.includes("not a valid FCM registration token")) {
